ghost/core/core/server/web/shared/middleware/max-limit-cap.js

```diff
@@ -0,0 +1,55 @@
+const config = require('../../../../shared/config');
+
+// Prior to Ghost 6.x we allowed any limit value, including 'all', but as sites
+// grew in size it led to performance issues and mis-use of the API.
+
+// After Ghost 6.x we only allow a max limit of 100. This middleware enforces
+// that limit by rewriting the limit parameter before it reaches any API code.
+
+// Temporary exceptions to the max limit rule
+const EXCEPTION_ENDPOINTS = [
+    '/ghost/api/admin/posts/export/',
+    '/ghost/api/admin/emails/' // /:id/batches/ and /:id/recipient-failures/
+];
+
+function maxLimitCap(req, res, next) {
+    // TODO: How expensive is reading from config? Maybe move outside of the function
+    const allowLimitAll = config.get('optimization:allowLimitAll') || false;
+    const maxLimit = config.get('optimization:maxLimit') || 100;
+
+    const limit = req.query.limit;
+
+    if (!limit) {
```

Comment:
If not removing the config reads, maybe move them below this line 🤔

Reply:
No, I was right that we couldn't override in tests when they were module-scope.  Added a workaround in https://github.com/TryGhost/Ghost/pull/24118/commits/cd0a383aa1cc97de3fd82f34c73560336d44124d which feels _ok_ although maybe a big magical 😅